### PR TITLE
Add TLS/mTLS settings for postgreSQL and Redis

### PR DESCRIPTION
**Pull request by Shivangi-ch**
This PR fixes: #28 

**Current behavior:-**
With the current settings we can use PostgreSQL as database and Redis as cache, we only have to specify it in the configurations.
However, the current `settings.py` does not specify any TLS/mTLS options for DB and cache.

**New Behavior:-**
So to support TLS connections with the database and cache I have added new options only for postgresql database and redis caches.
I have added new keys  such as sslmode, sslrootcert etc.
These values should be set in the configurations before running the server if we need to run in TLS/mTLS mode.

### CODE REVIEW

**Comment by Shivangi-ch** [comment]
Hey @hongquan,
I have taken care of  the recommendations  mentioned by you in PR #46
Also I have moved the TLS config  building logic to a separate file `tls_settings.py` and imported it in the main `settings.py`

Could you please review the PR

**Review by hongquan — CHANGES_REQUESTED** [review]
Please rename _tls_settings.py_ to _settings_helpers.py_. Later on, we can move some other code to it.

Please format your code with  [`Ruff`](https://github.com/astral-sh/ruff). Only format your new file. Don't touch other files (this must be done in a dedicated PR somedays in the future).

**Comment by Shivangi-ch** [comment]
> Please rename _tls_settings.py_ to _settings_helpers.py_. Later on, we can move some other code to it.

@hongquan I have renamed the file as suggested.

> Please format your code with [`Ruff`](https://github.com/astral-sh/ruff). Only format your new file. Don't touch other files (this must be done in a dedicated PR somedays in the future).

and formatted the new file using ruff.

File: src/pretix/settings_helpers.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    db_ssl_mode = config.get("database", "sslmode", fallback="disable")
+    # add postgresql TLS options
+    if db_ssl_mode != "disable" and db_backend == "postgresql":
+        db_tls_config = {
+            "sslmode": db_ssl_mode,
+            "sslrootcert": config.get("database", "sslrootcert"),
+        }
+        # add postgresql mTLS options
+        if config.has_option("database", "sslcert"):
+            db_tls_config.update(
```
Comment (hongquan) [inline src/pretix/settings_helpers.py]: I thinks in these lines, writing like this is clearer:

```py
db_tls_config['sslcert'] = config.get('database', 'sslcert')
```

**Review by hongquan — CHANGES_REQUESTED** [review]

File: src/pretix/settings.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import os
 import sys
 from urllib.parse import urlparse
-
+from .settings_helpers import *
```
Comment (hongquan) [inline src/pretix/settings.py]: Please not use `import *`. Be explicit about what you want to import.
[PEP8 coding style](https://peps.python.org/pep-0008/#imports)

**Review by hongquan — CHANGES_REQUESTED** [review]

File: src/pretix/settings.py [hunk 3]
```diff
@@ -209,22 +214,28 @@
...
-    CACHES['redis'] = {
-        "BACKEND": "django_redis.cache.RedisCache",
-        "LOCATION": config.get('redis', 'location'),
-        "OPTIONS": {
+    OPTIONS = {
             "CLIENT_CLASS": "django_redis.client.DefaultClient",
             "REDIS_CLIENT_KWARGS": {"health_check_interval": 30}
         }
+    redis_tls_config = build_redis_tls_config(config)
+    if(redis_tls_config is not None):
```
Comment (hongquan) [inline src/pretix/settings.py]: You didn't format your code with Ruff?

**Review by hongquan — CHANGES_REQUESTED** [review]

File: src/pretix/settings.py (hunk 3 above)
Comment (hongquan) [reply src/pretix/settings.py]: Ah, this file is an existing file, we don't run Ruff on it. Then you have to format your code manually, based on PEP8 coding style.

File: src/pretix/settings_helpers.py (hunk 1 above)
Comment (Shivangi-ch) [reply src/pretix/settings_helpers.py]: I have made the changes.

File: src/pretix/settings.py (hunk 2 above)
Comment (Shivangi-ch) [reply src/pretix/settings.py]: I have made these changes.

File: src/pretix/settings.py (hunk 3 above)
Comment (Shivangi-ch) [reply src/pretix/settings.py]: formatted the code.

File: src/pretix/settings.py [hunk 4]
```diff
@@ -209,22 +214,28 @@
 
 HAS_REDIS = config.has_option('redis', 'location')
 if HAS_REDIS:
+    OPTIONS = {
```
Comment (hongquan) [inline src/pretix/settings.py]: Please rename it to `redis_options`. We don't expose it as application settings, no reason to make it all-uppercase.

**Review by hongquan — CHANGES_REQUESTED** [review]

File: src/pretix/settings.py (hunk 4 above)
Comment (Shivangi-ch) [reply src/pretix/settings.py]: Renamed to `redis_options`.

**Review by hongquan — APPROVED** [review]

**Comment by mariobehling** [comment]
Due to challenges with the previous branch merges, it would be easiest if you could make this PR again against the current development branch. Appreciate your effort!

**Pull request merged**